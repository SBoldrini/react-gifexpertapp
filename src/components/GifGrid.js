import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category );



  return (
    <>
      <h3 id="grid-title" className="animate__animated animate__fadeIn">{ category.toUpperCase() }</h3>
      <hr />

      { loading && <p className="animate__animated animate__flash">Loading...</p> }

      <div className="card-grid">
        {
          images.map(img => (
            <GifGridItem
              key = { img.id }
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
