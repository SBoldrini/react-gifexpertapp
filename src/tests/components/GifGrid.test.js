import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba del componente <GifGrid />', () => {
  
  const category = 'Pokemon';
  
  test('Debe de mostrar el componente correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true, 
    });
    
    const wrapper = shallow(<GifGrid category={ category }/>);
    
    expect(wrapper).toMatchSnapshot();
    
  });
  
  
  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    
    const gifs = [{
      id: 'ABC',
      url: 'https://unapureba.com/prueba.jpg',
      title: 'Titulo de Test',
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false, 
    });
    
    const wrapper = shallow(<GifGrid category={ category }/>);

    // expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('p').exists()).toBe(false);

    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
  


  
});
