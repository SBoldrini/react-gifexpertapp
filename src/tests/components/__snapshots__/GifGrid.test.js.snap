// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Pokemon
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
